OverlayPlugin.Core/resources/opcodes.json

```diff
@@ -0,0 +1,44 @@
+{
+    "2022.05.27.0000.0000": {
```

Comment:
This is not a blocking question and more just a thought for the future.  I think it would be nice to have OverlayPlugin check for remote opcodes.json updates (periodically) and fall back to opcodes.json on the filesystem.  That way there's not pressure to push a release for every release.  However, how do we determine if this file has changed? Dates? Hash? An explicit version number <_<? Check once per day??

Reply:
I didn't want to tackle something like that with this PR because it would have made this PR way bigger, and I'm not as experienced with that process and how it might e.g. impact ACT's autoupdater functionality for OP due to overwriting the file or not.

I agree that a check-for-updates would be nice but I think it can be added cleanly to what's already there once the mechanisms have been decided on.

Reply:
Had some offline discussions with @quisquous about this, but there's a decent amount that can be done for varying degrees of control/automation. I think there's a very simple mechanism that can be a GitHub action for "when opcodes.json is touched, run some script to make it hosted in GitHub pages. Check last-modified header and if that's newer either straight up just use it (without downloading/updating locally) or update my local opcodes." There's some level of concern about what does a failure look like, but maybe it's not a huge deal. In the non-download scenario, you have users running out of GitHub itself (as long as they can 200 GET then you're good) and if there's any issues the user is pretty likely to just close and reopen ACT as a first line of defense. This state would exist until the next release. In the download scenario, there'd be less GitHub calls, but seeing as we're playing an MMO and GitHub should have decent uptime I don't really see this being a huge issue; I'm slightly more afraid of Windows file swapping issues, but that might be personal bias and ignorance.

However, there is some level of concern that this is now downloading something/doing something remote that the user didn't sign up for. I don't personally think this is a huge issue, but people might find fault. The updater is already reading the GitHub patch notes and putting them in front of you, it's pretty similar at that point.

There is another option I brought up which is simply an auto-update flag. I think most users don't care about anything that's in OverlayPlugin and a changelog link can be added to the plugin's main landing page. This would alleviate the above issue, but would put more pressure to release more. This in turn would probably turn into some mechanism where each PR checks if there's an open pre-release and updates that accordingly if so, and then one of the maintainers would need to come along and upgrade the release from pre-release. This doesn't really help the issue that _someone_ needs to do it, but is probably a strong middle ground for these changes. However, the issue of the opcodes isn't _exactly_ solved in this state, but maybe is "good enough" if there's a moral dissent against the remote GET option.

A little more hacky, maybe the shortest term solution to the above (again, if remote GET isn't an option or is maybe too error prone), just have GitHub bots bump the version and release any time there's an opcodes.json change pushed to main. This is a little more vulnerable, since now you're squeezing in more changes that maybe you previously weren't sure about and a human isn't doing "validations", but I think the velocity of this repo is probably currently at its peak and this is probably a fine enough change as well that when combined with the auto-updater option is pretty invisible to all parties.

Reply:
Could you bump this into an issue of its own so the discussion isn't buried?

Reply:
Would having it automatically update when it sees that the game version is not found in the opcodes file work? I think the case that that doesn't handle is if someone gets the opcode wrong, pushes an update with it, but since that would be uncommon, you could just manually push a release with the fixed code.

Reply:
Created in https://github.com/OverlayPlugin/OverlayPlugin/issues/65, please feel free to subscribe yourselves as you all see fit; I don't know exactly who all is interested enough and didn't feel like cc'ing people individually as that conversation spins up and this one winds down.